Panic in actor placement.go

In this [workflow run](https://github.com/dapr/dapr/actions/runs/1276467499), the pubsub routing test fails because at the same time, `daprd` crashes with this error.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x139fad7]

goroutine 282 [running]:
github.com/dapr/dapr/pkg/actors/internal.(*ActorPlacement).Start.func4.2(0xc0005a49a0, 0x368e75f, 0xc000c98f38)
	/home/runner/work/dapr/dapr/src/github.com/dapr/dapr/pkg/actors/internal/placement.go:258 +0x97
github.com/dapr/dapr/pkg/actors/internal.(*ActorPlacement).Start.func4()
	/home/runner/work/dapr/dapr/src/github.com/dapr/dapr/pkg/actors/internal/placement.go:259 +0x1b8
created by github.com/dapr/dapr/pkg/actors/internal.(*ActorPlacement).Start
	/home/runner/work/dapr/dapr/src/github.com/dapr/dapr/pkg/actors/internal/placement.go:216 +0x105
```

Possibly related to #3726

RELEASE NOTE: **FIX** Fixed panic sending on a nil gRPC stream in the placement heartbeat loop.

--- a/pkg/actors/internal/placement.go
+++ b/pkg/actors/internal/placement.go
@@ -255,7 +255,11 @@ func (p *ActorPlacement) Start() {
 				p.clientLock.RLock()
 				defer p.clientLock.RUnlock()
 
-				err = p.clientStream.Send(&host)
+				// p.clientStream is nil when the app is unhealthy and
+				// daprd is disconnected from the placement service.
+				if p.clientStream != nil {
+					err = p.clientStream.Send(&host)
+				}
 			}()
 
 			if err != nil {

--- a/tests/apps/pubsub-subscriber_grpc/app.go
+++ b/tests/apps/pubsub-subscriber_grpc/app.go
@@ -84,6 +84,9 @@ func main() {
 }
 
 func initializeSets() {
+	lock.Lock()
+	defer lock.Unlock()
+
 	receivedMessagesA = sets.NewString()
 	receivedMessagesB = sets.NewString()
 	receivedMessagesC = sets.NewString()
@@ -115,6 +118,9 @@ func (s *server) OnInvoke(ctx context.Context, in *commonv1pb.InvokeRequest) (*c
 }
 
 func (s *server) getReceivedMessages() []byte {
+	lock.Lock()
+	defer lock.Unlock()
+
 	resp := receivedMessagesResponse{
 		ReceivedByTopicA:   receivedMessagesA.List(),
 		ReceivedByTopicB:   receivedMessagesB.List(),

